File: 482. License Key Formatting.java
class Solution {
    public String licenseKeyFormatting(String S, int K) {
        char[] ca = S.replaceAll("-", "").toUpperCase().toCharArray();
        StringBuilder sb = new StringBuilder();
        int left = ca.length % K;
        int count = 0;
        for (int i=0; i<ca.length; i++) {
            if (left != 0 && i == left) {sb.append('-'); count = 0;}
            if (count==K) {
                sb.append('-');
                count = 0;
            }
            sb.append(ca[i]);
            count++;
        }
        return sb.toString();
    }
}

File: 482. License Key Formatting.py
class Solution:
    def licenseKeyFormatting(self, S: str, K: int) -> str:
        S = S.replace("-","").upper()
        length = len(S)
        list_k = [S[0:len(S)% K]]
        S = S[len(S) % K:]
        i = 0
        while i < length:
            list_k.append(S[i:i+K])
            i += K
        list_k = [x for x in list_k if x]
        return "-".join(list_k)